list/list.js

```diff
@@ -261,15 +262,15 @@ var DefineList = Construct.extend("DefineList",
 			}
 			// otherwise we are setting multiple
 			else {
-				if (isArray(prop)) {
+				if (canReflect.isListLike(prop)) {
 					if (value) {
-						this.replace(prop);
+						canReflect.setValue(this, prop);
```

Comment:
though I wonder about `.splice` ... we might need something for that in can-reflect